src/Features/LanguageServer/ProtocolUnitTests/SemanticTokens/SemanticTokensRangesTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.SemanticTokens
+{
+    public class SemanticTokensRangesTests : AbstractSemanticTokensTests
+    {
+        public SemanticTokensRangesTests(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        {
+        }
+
+        [Theory, CombinatorialData]
+        public async Task TestGetSemanticTokensRanges_FullDocAsync(bool mutatingLspWorkspace, bool isVS)
+        {
+            var markup =
+@"{|caret:|}// Comment
+static class C { }
+";
+            await using var testLspServer = await CreateTestLspServerAsync(
+                markup, mutatingLspWorkspace, GetCapabilities(isVS));
+
+            var range = new LSP.Range { Start = new Position(0, 0), End = new Position(2, 0) };
+            var ranges = new LSP.Range[1] { range };
```

Comment:
Would be good to test with at least two distinct ranges so that the stitch method has coverage

Reply:
sounds good, I think we have a test already on the razor side, will bring that over